src/main/java/org/gridsuite/modification/modifications/byfilter/ByFormulaModification.java

```diff
@@ -99,13 +101,19 @@ protected String getNewValue(Identifiable<?> equipment, AbstractAssignmentInfos
         return applyOperation(formulaInfos.getOperator(), value1, value2).toString();
     }
 
+    static final int MAX_SCALE = 10;
+
     private Double applyOperation(Operator operator, @Nonnull Double value1, @Nonnull Double value2) {
+        BigDecimal bValue1 = BigDecimal.valueOf(value1);
+        BigDecimal bValue2 = BigDecimal.valueOf(value2);
+
         return switch (operator) {
-            case ADDITION -> value1 + value2;
-            case SUBTRACTION -> value1 - value2;
-            case MULTIPLICATION -> value1 * value2;
-            case DIVISION -> value1 / value2;
-            case PERCENTAGE -> (value1 / 100) * value2;
+            case ADDITION -> bValue1.add(bValue2).doubleValue();
+            case SUBTRACTION -> bValue1.subtract(bValue2).doubleValue();
+            case MULTIPLICATION -> bValue1.multiply(bValue2).doubleValue();
+            case DIVISION -> bValue1.divide(bValue2, MAX_SCALE, RoundingMode.HALF_EVEN).doubleValue();
+            case PERCENTAGE -> bValue1.divide(BigDecimal.valueOf(100.0), MAX_SCALE, RoundingMode.HALF_EVEN)
+                .multiply(bValue2).doubleValue();
```

Comment:
```suggestion
static final int MAX_SCALE = 10;
    public static double round(double value) {
        BigDecimal bd = BigDecimal.valueOf(value);
        bd = bd.setScale(MAX_SCALE, RoundingMode.HALF_UP);
        return bd.doubleValue();
    }

    private Double applyOperation(Operator operator, @Nonnull Double value1, @Nonnull Double value2) {
        return switch (operator) {
            case ADDITION -> round(value1 + value2);
            case SUBTRACTION -> round(value1 - value2);
            case MULTIPLICATION -> round(value1 * value2);
            case DIVISION -> round(value1 / value2);
            case PERCENTAGE -> round((value1 / 100) * value2);
        };
    }
```
I think this would be better because there is one less `BigDecimal.valueOf` and round is standardised (and may be transfered in an utility class)

Reply:
I see 2 issues :
 - first : the result is converted after the operation that can lead to precision error
 - second : we round for each operation so we did not really make this ligher maybe only for division